onnxruntime/python/tools/transformers/fusion_attention.py

```diff
@@ -458,6 +458,16 @@ def fuse(self, normalize_node, input_name_to_nodes, output_name_to_node):
                 if child.op_type == "LayerNormalization":
                     root_input = child.output[0]
 
+        children = input_name_to_nodes[root_input]
+        children_types = [child.op_type for child in children]
+        if children_types.count("MatMul") != 3:
```

Comment:
Add some comments for this block of code like example graph pattern.

Reply:
Done, thanks
I also feel like we can remove [these lines of code](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/python/tools/transformers/fusion_attention.py#L427-L435) completely, since this case with multiple Add consumers is handled by SkipLayerNorm fusion itself since [this commit](https://github.com/microsoft/onnxruntime/commit/7ed8bd4f95977d7601250564e3f40039d176a2c7#diff-2c80d28126f62441ef0d93ae1c9021a1c771d24bb9d4737feb606f45dd45c1ff)

Reply:
From my understanding, this code is to find root_input for SkipLayerNormalization.

It may be better to move to a place that computes root_input.

Reply:
It is not about normalize_node, but about its parent. When its parent is usual SkipLayerNorm, it has the same output for Attention and this normalize node, but when it is a SkipLayerNorm with multiple Add consumers, it has two outputs - one goes to Attention, and the other goes to normalize_node. In this case we need to pick the other SkipLayerNorm output.

Actually, having looked at this once again, I rewrote this part of code so that it is more explicit and concise, thanks